File: vueapp/src/library/plus/map-open.js
import Toast from '@/library/ui/ui-toast'

/* 打开地图 */
export default (dst,src,address)=>{
  try{
    dst = new plus.maps.Point(dst[0],dst[1]);
    src = new plus.maps.Point(src[0],src[1]);
    if(plus.os.name=='iOS'){
      plus.maps.openSysMap(dst,address,src);
    }else if(plus.os.name=='Android'){
      plus.maps.openSysMap(src,address,dst);
    }
  }catch(e){
    return Toast('请在APP内使用!');
  }
}

File: flutter/lib/library/plus/map-open.dart
import 'package:flutter/material.dart';
import 'package:webmis/library/plus/app-info.dart';
import 'package:url_launcher/url_launcher.dart';
import 'package:webmis/library/ui/ui-toast.dart';

/* 地图-打开导航 */
class MapOpen {
  static Future amap(BuildContext context, {String lon: '102.703689', String lat:'25.048474', String poiname: ''}) async {
    Map res = await appInfo();
    final String platform = res['platform']=='Android'?'android':'ios';
    String url = '${platform}amap://navi?sourceApplication=appname&amp;poiname=$poiname&amp;lat=$lat&amp;lon=$lon&amp;dev=0&amp;style=2';
    if(await canLaunch(url)){
      await launch(url);
    }else{
      Toast(context, '请安装高德地图!');
    }
  }
}